pkg/agent/util/net_linux.go

```diff
@@ -237,3 +242,170 @@ func DeleteOVSPort(brName, portName string) error {
 	cmd := exec.Command("ovs-vsctl", "--if-exists", "del-port", brName, portName)
 	return cmd.Run()
 }
+
+func HostInterfaceExists(ifName string) bool {
+	_, err := netlink.LinkByName(ifName)
+	if err == nil {
+		return true
+	}
+	if _, ok := err.(netlink.LinkNotFoundError); ok {
+		return false
+	}
+	klog.ErrorS(err, "Failed to find host interface", "name", ifName)
+	return false
+}
+
+func GetUplinkConfig(uplinkIfName string) (*config.AdapterNetConfig, error) {
+	iface, err := net.InterfaceByName(uplinkIfName)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get interface", "uplinkIfName", uplinkIfName)
+		return nil, err
```

Comment:
Generally we should either return the error (with more context prepended) or log it but not do both of them, otherwise eventually it would cause duplicate errors, which may be confusing. This patch has a few such duplicate error handling.